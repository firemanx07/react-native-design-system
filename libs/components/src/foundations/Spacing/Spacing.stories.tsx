import { Story } from '@storybook/react';
import React from 'react';

import { Spacing as SpacingBase } from './Spacing';

export default {
  component: SpacingBase,
  title: 'Foundations/Spacing',
};

const Template: Story = () => <SpacingBase />;

const parameters = {
  controls: {
    include: [],
  },
};

const defaultArgs = {};

export const Spacing = Template.bind({});

Spacing.parameters = parameters;
Spacing.args = {
  ...defaultArgs,
};
