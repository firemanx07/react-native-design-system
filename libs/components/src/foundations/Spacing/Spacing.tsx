import { styled, useNamespacedTheme } from '@proxym/themes';
import React, { useCallback, useMemo, Fragment } from 'react';

import { BaseText, TextVariant } from '../../atoms';

export const Spacing = () => {
  const { spacing } = useNamespacedTheme();

  const spacingKeys = useMemo(
    () =>
      Object.values(spacing).filter(
        (value: string | number) => typeof value === 'string',
      ) as string[],
    [],
  );

  const renderSpace = useCallback((key: any) => {
    return (
      <Fragment key={key}>
        <BaseText variant={TextVariant.M}>{key}</BaseText>
        <Space height={spacing[key] as unknown as number} />
      </Fragment>
    );
  }, []);

  return <Container>{spacingKeys.map(renderSpace)}</Container>;
};

const Container = styled.View`
  padding: ${({ theme }) => theme.ds.spacing.mediumSmall}px;
  flex: 1;
  background-color: ${({ theme }) => theme.ds.colors.graySoft};
  width: 400px;
`;

type SpaceType = {
  height: number;
};

const Space = styled.View<SpaceType>`
  background-color: ${({ theme }) => theme.ds.colors.light};
  margin-bottom: ${({ theme }) => theme.ds.spacing.primary}px;
  height: ${({ height }) => height}px;
`;

export default Spacing;
