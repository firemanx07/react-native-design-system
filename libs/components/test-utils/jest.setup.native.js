import { Animated } from 'react-native';

require('react-native-reanimated/lib/reanimated2/jestUtils').setUpTests();
require('react-native-gesture-handler/jestSetup.js');

console.warn = () => {
  //this is mock
};

Animated['timing'] = (value, config) => ({
  start: callback => {
    value.setValue(config.toValue);
    callback && callback({ finished: true });
  },
  stop: jest.fn(),
});
