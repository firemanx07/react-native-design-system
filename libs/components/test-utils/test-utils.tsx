import { theme, ThemeProvider } from '@celebrate-app/themes';
import { Story } from '@storybook/react';
import {
  cleanup,
  render,
  RenderOptions,
  waitFor,
} from '@testing-library/react-native';
import React, { ReactNode } from 'react';

const Providers = ({ children }: { children: ReactNode }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

export const customRender = (
  ui: React.ReactElement<any>,
  options?: RenderOptions
) => {
  return render(ui, { wrapper: Providers, ...options });
};

beforeEach(cleanup);
afterEach(cleanup);

export function testStory(displayName: string, TestComponent: Story) {
  test(`${displayName} snapshot test`, async () => {
    const comp = customRender(<TestComponent {...TestComponent.args} />);
    jest.runOnlyPendingTimers();

    await waitFor(() => {
      expect(comp).toBeTruthy();
      expect(comp.toJSON()).toMatchSnapshot();
    });
  });
}

export function testComponentSnapshot(
  TestComponent: React.ReactElement<
    any,
    string | React.JSXElementConstructor<any>
  >
) {
  test('Snapshot test', () => {
    const comp = customRender(TestComponent);
    expect(comp).toBeTruthy();
    expect(comp.toJSON()).toMatchSnapshot();
  });
}

export function testAllStories(stories: any) {
  Object.keys(stories).forEach((story) => {
    if (story !== 'default') {
      testStory(story, stories[story]);
    }
  });
}
