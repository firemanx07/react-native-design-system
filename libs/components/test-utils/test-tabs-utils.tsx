import { waitFor } from '@testing-library/react-native';

import { customRender } from './test-utils';

type renderTabsProps = {
  elem: JSX.Element;
  testIDs: string[];
};
export const renderTabs = ({ elem, testIDs }: renderTabsProps) => {
  const comp = customRender(elem);
  const tabs: ReturnType<typeof comp.getByTestId>[] = [];
  testIDs.forEach(testID => tabs.push(comp.getByTestId(testID)));
  return { comp, tabs };
};
export const testTabs = async (args: ReturnType<typeof renderTabs>) => {
  await waitFor(() => {
    expect(args.comp).toBeTruthy();
    args.tabs.forEach(tab => expect(tab).toBeDefined());
  });
};
