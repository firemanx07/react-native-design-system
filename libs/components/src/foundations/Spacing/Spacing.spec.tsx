import { testAllStories } from '../../../test-utils';
import * as stories from './Spacing.stories';

describe('Spacing tests', () => {
  testAllStories(stories);
});
