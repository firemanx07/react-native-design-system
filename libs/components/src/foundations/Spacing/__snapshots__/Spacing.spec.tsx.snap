// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spacing tests Spacing snapshot test 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#DEDFE0",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingBottom": 20,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 20,
        "width": 400,
      },
    ]
  }
>
  <Text
    fontFamily="Rubik-Regular"
    fontSize={14}
    lineHeight={20}
    style={
      [
        {
          "color": "#2C2C2C",
          "fontFamily": "Rubik-Regular",
          "fontSize": 14,
          "lineHeight": 20,
        },
      ]
    }
  >
    tiny
  </Text>
  <View
    height={4}
    style={
      [
        {
          "backgroundColor": "#FFFFFF",
          "height": 4,
          "marginBottom": 16,
        },
      ]
    }
  />
  <Text
    fontFamily="Rubik-Regular"
    fontSize={14}
    lineHeight={20}
    style={
      [
        {
          "color": "#2C2C2C",
          "fontFamily": "Rubik-Regular",
          "fontSize": 14,
          "lineHeight": 20,
        },
      ]
    }
  >
    small
  </Text>
  <View
    height={8}
    style={
      [
        {
          "backgroundColor": "#FFFFFF",
          "height": 8,
          "marginBottom": 16,
        },
      ]
    }
  />
  <Text
    fontFamily="Rubik-Regular"
    fontSize={14}
    lineHeight={20}
    style={
      [
        {
          "color": "#2C2C2C",
          "fontFamily": "Rubik-Regular",
          "fontSize": 14,
          "lineHeight": 20,
        },
      ]
    }
  >
    primarySmall
  </Text>
  <View
    height={12}
    style={
      [
        {
          "backgroundColor": "#FFFFFF",
          "height": 12,
          "marginBottom": 16,
        },
      ]
    }
  />
  <Text
    fontFamily="Rubik-Regular"
    fontSize={14}
    lineHeight={20}
    style={
      [
        {
          "color": "#2C2C2C",
          "fontFamily": "Rubik-Regular",
          "fontSize": 14,
          "lineHeight": 20,
        },
      ]
    }
  >
    primary
  </Text>
  <View
    height={16}
    style={
      [
        {
          "backgroundColor": "#FFFFFF",
          "height": 16,
          "marginBottom": 16,
        },
      ]
    }
  />
  <Text
    fontFamily="Rubik-Regular"
    fontSize={14}
    lineHeight={20}
    style={
      [
        {
          "color": "#2C2C2C",
          "fontFamily": "Rubik-Regular",
          "fontSize": 14,
          "lineHeight": 20,
        },
      ]
    }
  >
    mediumSmall
  </Text>
  <View
    height={20}
    style={
      [
        {
          "backgroundColor": "#FFFFFF",
          "height": 20,
          "marginBottom": 16,
        },
      ]
    }
  />
  <Text
    fontFamily="Rubik-Regular"
    fontSize={14}
    lineHeight={20}
    style={
      [
        {
          "color": "#2C2C2C",
          "fontFamily": "Rubik-Regular",
          "fontSize": 14,
          "lineHeight": 20,
        },
      ]
    }
  >
    medium
  </Text>
  <View
    height={24}
    style={
      [
        {
          "backgroundColor": "#FFFFFF",
          "height": 24,
          "marginBottom": 16,
        },
      ]
    }
  />
  <Text
    fontFamily="Rubik-Regular"
    fontSize={14}
    lineHeight={20}
    style={
      [
        {
          "color": "#2C2C2C",
          "fontFamily": "Rubik-Regular",
          "fontSize": 14,
          "lineHeight": 20,
        },
      ]
    }
  >
    large
  </Text>
  <View
    height={28}
    style={
      [
        {
          "backgroundColor": "#FFFFFF",
          "height": 28,
          "marginBottom": 16,
        },
      ]
    }
  />
  <Text
    fontFamily="Rubik-Regular"
    fontSize={14}
    lineHeight={20}
    style={
      [
        {
          "color": "#2C2C2C",
          "fontFamily": "Rubik-Regular",
          "fontSize": 14,
          "lineHeight": 20,
        },
      ]
    }
  >
    extraLarge
  </Text>
  <View
    height={32}
    style={
      [
        {
          "backgroundColor": "#FFFFFF",
          "height": 32,
          "marginBottom": 16,
        },
      ]
    }
  />
  <Text
    fontFamily="Rubik-Regular"
    fontSize={14}
    lineHeight={20}
    style={
      [
        {
          "color": "#2C2C2C",
          "fontFamily": "Rubik-Regular",
          "fontSize": 14,
          "lineHeight": 20,
        },
      ]
    }
  >
    huge
  </Text>
  <View
    height={40}
    style={
      [
        {
          "backgroundColor": "#FFFFFF",
          "height": 40,
          "marginBottom": 16,
        },
      ]
    }
  />
</View>
`;
